tests/tracing/test_dnatracing_methods.py

```diff
@@ -386,3 +387,392 @@ def test_crop_array(bounding_box: tuple, target: np.array, pad_width: int) -> No
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0],
+                    [0, 0, 0, 0, 0],
+                ]
+            ),
+        ),
+        # Complex line without smoothing, unchanged.
+        (
+            np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                ]
+            ),
+            5.0,
+            False,
+            0.0,
+            np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                ]
+            ),
+        ),
+        # Complex line with smoothing, smoothed out.
+        (
+            np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                ]
+            ),
+            5.0,
+            False,
+            5.0,
+            np.array(
+                [
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                    [0, 1, 1, 0, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 1, 0, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 0, 1, 0],
+                    [0, 0, 0, 1, 0, 1, 0, 0],
+                    [0, 0, 0, 0, 1, 1, 0, 0],
+                    [0, 0, 0, 0, 1, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0, 0, 0],
+                ]
+            ),
+        ),
+    ],
+)
+def test_get_splined_traces(
```

Comment:
This test is doing a lot of work. Some of the steps could I think be moved out and a separate fixture for this specific test.

For example the instantiated `dnatrace` fixture could be setup with....

```
@pytest.fixture()
def dnatrace_spline(dnatrace: dnaTrace) -> dnaTrace:
   """dnaTrace object for splining tests."""
   dnatrace.pixel_to_nm_scaling = 1.0
   dnatrace.n_grain = 1
   return dnatrace
```

...and then use `dnatrace_spline`.

That said I don't think much else could be broken out as the reordering and other steps need performing conditional on the skeleton that is being passed in.

Has made me realise that the refactoring I did is incomplete as there aren't tests for things like `reordertrace.linearTrace()` or `reordertrace.circularTrace()`, captured in #706.

Reply:
I've added the fixture, but I likewise cannot think of a way to break out the logic for this rather messy test ☹️